src/dotnet/CommandLine/CommandLineApplication.cs

```diff
@@ -64,7 +64,21 @@ public CommandLineApplication Command(string name, bool throwOnUnexpectedArg = t
         public CommandLineApplication Command(string name, Action<CommandLineApplication> configuration,
             bool throwOnUnexpectedArg = true)
         {
-            var command = new CommandLineApplication(throwOnUnexpectedArg) { Name = name, Parent = this };
+            var command = new CommandLineApplication(throwOnUnexpectedArg) { Name = name };
+            return Command(command, configuration, throwOnUnexpectedArg);
+        }
+
+        public CommandLineApplication Command(CommandLineApplication command, bool throwOnUnexpectedArg = true)
+        {
+            return Command(command, _ => { }, throwOnUnexpectedArg);
+        }
+
+        public CommandLineApplication Command(
+            CommandLineApplication command,
+            Action<CommandLineApplication> configuration,
+            bool throwOnUnexpectedArg = true)
```

Comment:
If you really feel strongly about this I can make the change. But what I'm saying is there has to be some discretion on when to add the null check vs when not too. We don't have all our methods doing null checks. And I felt this particular one doesn't need it.